src/rebar_pkg_resource.erl

```diff
@@ -206,17 +206,19 @@ store_etag_in_cache(Path, ETag) ->
 %%%=============================================================================
 %%% Private functions
 %%%=============================================================================
--spec cached_download(TmpDir, CachePath, Pkg, ETag, ETagPath, UpdateETag) -> Res when
+-spec cached_download(TmpDir, CachePath, Pkg, State, ETag, ETagPath, UpdateETag) -> Res when
       TmpDir :: file:name(),
       CachePath :: file:name(),
       Pkg :: package(),
+      State :: rebar_state:t(),
       ETag :: binary(),
       ETagPath :: file:name(),
       UpdateETag :: boolean(),
       Res :: ok | {unexpected_hash, integer(), integer()} | {fetch_fail, binary(), binary()}.
-cached_download(TmpDir, CachePath, Pkg={pkg, Name, Vsn, _Hash, RepoConfig}, ETag,
+cached_download(TmpDir, CachePath, Pkg={pkg, Name, Vsn, _Hash, RepoConfig}, State, ETag,
                 ETagPath, UpdateETag) ->
-    case request(RepoConfig, Name, Vsn, ETag) of
+    CDN = rebar_state:get(State, rebar_packages_cdn),
+    case request(RepoConfig#{repo_url => rebar_utils:to_binary(CDN)}, Name, Vsn, ETag) of
```

Comment:
Should the `to_binary` call be made when setting the config so it's always got the right type, rather than converting at the call-site? I'm not really sure either way since I guess we could consider the URL typing (strict binary instead of iodata()) to be an implementation detail.

Reply:
Yeah, I almost did that and can do and I think it's ok since this value isn't used anywhere else. Implementation detail is why I didn't though, but once again, I think it's fine. Will change.